### ISSUE TITLE
Cannot open extension popup

### ISSUE THREAD

**Issue by lesleyxyz** [tubearchivist/browser-extension#48]
When clicking the extension icon, it isn't showing the popup. It just hangs in this state:

![image](https://github.com/user-attachments/assets/04a5ba0a-397c-4b30-9b31-a4c9251d1b28)

I tried removing & reinstalling the extension.
Also tried in incognito where this is the only extension enabled.

Key | Value
-- | --
Google Chrome | 130.0.6723.116 (Official Build) (64-bit) (cohort: M130 Rollout)
Revision | 6ac35f94ae3d01152cf1946c896b0678e48f8ec4-refs/branch-heads/6723@{#1764}
OS | Windows 11 Version 24H2 (Build 26100.2033)
JavaScript | V8 13.0.245.18
User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
Command Line | "C:\Program Files\Google\Chrome\Application\chrome.exe" --flag-switches-begin --flag-switches-end --no-startup-window /prefetch:5 --flag-switches-begin --flag-switches-end

**Comment by bbilly1** [comment]
I think for others it was because of an other extension blocking something. You could try to deactivate all other extensions.

Also sometimes it can just take a few seconds until it establishes contact with the server, that's not fully async there it looks like...

**Comment by lesleyxyz** [comment]
I can reproduce this with only this extension enabled.

However, after waiting 15 - 20s it finally opens.
This happens everytime.

If this is due to establishing connection, I believe it would be nicer UX if the popup would open and the connection would happen async (not blocking the UI).

Of course this would be just nice-to-have :)

**Comment by bbilly1** [comment]
> If this is due to establishing connection, I believe it would be nicer UX if the popup would open and the connection would happen async (not blocking the UI).

Agree, I wasn't able to spend much time on that. Please help implement that and open a PR.

**Comment by BaturhanKahraman** [comment]
it is still active. I can not open it either.
